src/server/auth/server/roles.go

```diff
@@ -172,22 +172,33 @@ func init() {
 	})
 
 	// Project related roles
+	projectViewerRole := registerRole(&auth.Role{
+		Name:          auth.ProjectViewerRole,
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_REPO},
```

Comment:
sorry for late reply. The semantics are the other way around. Reading over the existing implementation, I think that it could be better and more clear. The `ResourceTypes` mean the Resources (auth levels) that this role can be applied to.

I thought that we were validating this at `[Create/Modify]RoleBinding` but turns out that we aren't...

As far as API surface it currently gets exposed in the semantics of `GetPermissions(resource) -> (permissions, roles)` where `roles` is all the roles the user has related to that resource that satisfied these permissions. So for example, if I have RepoReader assigned to me at the CLUSTER auth level, I should get RepoReader back in the response of `GetPermissions("repoFoo", "user:alon") -> (REPO_READ, RepoReader)`